### ISSUE TITLE
Changing the Haskell suggestion 

### ISSUE THREAD

**Issue by aryzach** [ossu/computer-science#669]
I really liked learning Haskell. But I had to do it differently than how it's introduced here. The 'Learn you a Haskell' is super long, and you have to read a lot of unnecessary content to learn about laziness and monads, and when you do, there's aren't any exercises to practice. This book is a good reference manual, but not a great introduction.

Here are two version of a Haskell class (lecture notes, then homework) that both cover laziness, purity, and monads. I'm learning a lot more from these classes than the book (but maybe that's partially because I've spent time reading the book before taking these classes too). Regardless, I think the class is a beneficial substitution:

https://www.seas.upenn.edu/~cis194/spring13/lectures.html
https://www.seas.upenn.edu/~cis194/fall16/

I've only taken about half of the first one, but I'm guessing their very similar.

**Comment by waciumawanjohi** [comment]
Adding context:
* The course you linked is UPenn's Introduction to Haskell.
* The Fall 2016 version of this course is currently included in our Extras/Courses
* The paid resource that the curriculum mentions is haskellbook.com. [The author of that book has a list of favorite Haskell learning resource](https://bitemyapp.com/blog/functional-education/). His top recommendation is this course. (He was writing in 2014, so recommends the 2013 version)
* I can find no convincing contrast between the two versions of the course.† Given that, it seems wiser to default to the more recently produced class.

I support this proposal!

†: I particularly enjoyed this snippet I found online:

> "CIS 194 course page has two publicly available incarnations to consider:
> https://www.seas.upenn.edu/~cis194/fall16/ (Joachim Breitner) https://www.seas.upenn.edu/~cis194/spring13/ (Brent Yorgey)
> Their variations should give ample room for contributors to disagree about particulars."

**Comment by waciumawanjohi** [comment]
I'm going to tag along with this issue to propose a further change:
### Haskell and Prolog should move from Core to Advanced.

[CS2013](https://www.acm.org/binaries/content/assets/education/cs2013_web_final.pdf) defines 18 knowledge areas that computer science students should learn. Within each, the document lays out which topics are absolutely core (Tier 1), which are very important (Tier 2, of which students should learn a minimum 80%) and which topics are elective.

Programming Languages is one of the 18 knowledge areas. Laziness and Monads fall into the topic _Advanced Programming Constructs_ which is an elective topic. Similarly _Logic Programming_ (including backtracking and unification) is an elective topic.

OSSU Core CS should not recommend courses that focus on elective topics. Currently Advanced CS states "Not every course from [an Advanced CS] subcategory needs to be taken". Therefore the best place for Haskell and Prolog is Advanced Programming.

**Comment by waciumawanjohi** [comment]
Will keep this issue open for comments until April 15.

**Comment by aryzach** [comment]
I would definitely agree with this. The READ ME suggests taking all these classes / books in order. I've never programmed before starting all this, and the Haskell and Prolog parts, while I really enjoyed them, I think slowed me down, and other areas probably should've been prioritized first. I haven't done everything in order exactly, but I was committed to finishing core CS before moving on. 

Also, thank you for all the work you contribute here. I've really liked this curriculum and being part of the MOOC community. I've been at it for over two years (all OSSU classes), and am now looking to apply for grad (research-focused) CS programs.

**Comment by krishnakumarg1984** [comment]
@waciumawanjohi 
>Given that, it seems wiser to default to the more recently produced class.

Well, going by your words, we should link to the [Spring '23 version ](https://www.seas.upenn.edu/~cis1940/spring23/schedule.html) for the ongoing class or the [Fall '18 version](https://www.seas.upenn.edu/~cis1940/fall18/) for the most recently completed semester available online.

**Comment by waciumawanjohi** [comment]
@krishnakumarg1984 Can you open a new issue with the new change that you'd like to propose? A one sentence comment on an issue closed three years ago is not very accessible to contributors.
